lms/djangoapps/grades/signals/handlers.py

```diff
@@ -124,19 +109,53 @@ def recalculate_subsection_grade_handler(sender, **kwargs):  # pylint: disable=u
     """
     if not settings.FEATURES.get('ENABLE_SUBSECTION_GRADES_SAVED', False):
         return
-    try:
-        course_id = kwargs.get('course_id', None)
-        usage_id = kwargs.get('usage_id', None)
-        student = kwargs.get('user', None)
 
-        course_key = CourseLocator.from_string(course_id)
-        usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
-
-        from lms.djangoapps.grades.new.subsection_grade import SubsectionGradeFactory
-        SubsectionGradeFactory(student).update(usage_key, course_key)
-    except Exception as ex:  # pylint: disable=broad-except
+    try:
+        course_id = kwargs['course_id']
+        usage_id = kwargs['usage_id']
+        student = kwargs['user']
+    except KeyError:
         log.exception(
-            u"Failed to process SCORE_CHANGED signal. "
-            "user: %s, course_id: %s, "
-            "usage_id: %s. Exception: %s", unicode(student), course_id, usage_id, ex.message
+            u"Failed to process SCORE_CHANGED signal, some arguments were missing."
+            "user: %s, course_id: %s, usage_id: %s.",
+            kwargs.get('user', None),
+            kwargs.get('course_id', None),
+            kwargs.get('usage_id', None),
         )
+        return
+
+    course_key = CourseLocator.from_string(course_id)
+    usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    manager = get_block_structure_manager(course_key)
+    block_structure = manager.get_collected()
+
+    subsections_to_update = block_structure.get_transformer_block_field(
+        usage_key,
+        GradesTransformer,
+        'containing_subsections',
+        set()
+    )
+
+    for subsection in subsections_to_update:
+        SubsectionGradeFactory(student).update(subsection, course_key)
+
+
+def connect_handlers():
+    """
+    Connect signal handlers to the appropriate signals
+
+    `dispatch_uid` prevents handlers from getting registered multiple times
+    """
+
+    SCORE_CHANGED.connect(
+        recalculate_subsection_grade_handler,
+        dispatch_uid=u'grades.recalculate_subsection_grade_handler'
+    )
+    score_set.connect(
+        submissions_score_set_handler,
+        dispatch_uid=u'grades.submissions_score_set_handler'
+    )
+    score_reset.connect(
+        submissions_score_reset_handler,
+        dispatch_uid=u'grades.submissions_score_reset_handler'
+    )
```

Comment:
I like the current implementation because of what Cliff alluded to above. I take that to mean that if we were to use the `@receiver` decorator here, we would still need to import the decorated function in the app config, but wouldn't be doing anything with it. So the import would be considered unused and make pylint mad.